File: calculator.py
import math
print("==========================================")
print("\n========== Scientific Calculator =========\n")
print("==========================================")

def acceleration():
    v = float(input("Enter Final Velocity: "))
    u = float(input("Enter Initial Velocity: "))
    t = float(input("Enter Time: "))
    a = ((v - u) / t)
    print(f"Acceleration = {a:.2f} ")


def final_velocity():
    u = float(input("Enter Initial Velocity: "))
    a = float(input("Enter Acceleration: "))
    t = float(input("Enter Time: "))
    v = (u + a * t)
    print(f"Final Velocity = {v:.2f} m/s")

def displacement():
    u = float(input("Enter Initial Velocity: "))
    t = float(input("Enter Time: "))
    a = float(input("Enter Acceleration: "))
    s = (u * t) + 0.5 * (a) * (t * t)
    print(f"Displacement = {s:.2f} meters")

def force():
    m = float(input("Enter Mass: "))
    a = float(input("Enter Acceleration: "))
    f = (m * a)
    print(f"Force = {f:.2f} newton")

def kinetic_energy():
    m = float(input("Enter Mass: "))
    v = float(input("Enter Final Velocity: "))
    ke = (0.5 * m * v * v)
    print(f"Kinetic Enerfy = {ke:.2f}")

def potential_energy():
    m = float(input("Enter Mass: "))
    h = float(input("Enter Height: "))
    g = float(input("Enter Gravitational Acceleration: "))
    pe = (m * h * g)
    print(f"Potential Energy = {pe:.2f}") 

def work():
    f = float(input("Enter Force: "))
    s = float(input("Enter Displacement: "))
    w = (f * s)
    print(f"Work = {w:.2f} joules")

def power():
    w = float(input("Enter Work: "))
    t = float(input("Enter Time: "))
    Po = (w / t)
    print(f"Power = {Po:.2f}")

def momentum():
    m = float(input("Enter Mass: "))
    v = float(input("Enter Final Velocity: "))
    p = (m * v)
    print(f"Momentum = {p:.2f} m/s")

def frequency():
    T = float(input("Enter Time Period: "))
    fre = (1 / T)
    print(f"Frequency = {fre:.4f} Hz")

def sin_theta():
    theta_degree = float(input("\nEnter Angle In Degrees: "))
    radian_degree = math.radians(theta_degree)

    Sintheta = math.sin(radian_degree)

    print(f"Sin {theta_degree}° = {Sintheta:.2f}")


def cos_theta():
    theta_degree = float(input("\nEnter Angle In Degrees: "))
    theta_rad = math.radians(theta_degree)
    Costheta = math.cos(theta_rad)
    print(f"{theta_degree}° = {Costheta:.2f}")

def tan_theta():
    deg_value = float(input("Enter Angle In Degrees: "))
    rad_value = math.radians(deg_value)
    Tantheta = math.tan(rad_value)
    print(f"{deg_value}° = {Tantheta:.2f}")

def expression():
    u = float(input("Enter Initial Velocity: "))
    a = float(input("Enter Acceleration: "))
    s = float(input("Enter Displacement: "))
    ex = (u * u) + ( 2 * a * s)
    eq = math.sqrt(ex)
    print(f" v^2 = u^2 + 2as ==> {eq:.2f}")

def main():
    while True:
        print("\n=== Main Menu ===\n")
        print("1. Acceleration\n")
        print("2. Final Velocity\n")
        print("3. Displacement\n")
        print("4. Force\n")
        print("5. Kinetic Energy\n")
        print("6. Potential Energy\n")
        print("7. Work\n")
        print("8. Power\n")
        print("9. Momentum\n")
        print("10. Frequency\n")
        print("11. Sin(theta)\n")
        print("12. Cos(theta)\n")
        print("13. Tan(theta)\n")
        print("14. Expression (v^2 = u^2 + 2as)\n")
        print("0. Exit\n")

        choice = input("Enter Your Choice (0 to Exit): ")

        if(choice == '1'):
            acceleration()
        elif(choice == '2'):
            final_velocity()
        elif(choice == '3'):
            displacement()
        elif(choice == '4'):
            force()
        elif(choice == '5'):
            kinetic_energy()
        elif(choice == '6'):
            potential_energy()
        elif(choice == '7'):
            work()
        elif(choice == '8'):
            power()
        elif(choice == '9'):
            momentum()
        elif(choice == '10'):
            frequency()
        elif(choice == '11'):
            sin_theta()
        elif(choice == '12'):
            cos_theta() 
        elif(choice == '13'):
            tan_theta()
        elif(choice == '14'):
            expression()                   
        elif(choice == '0'):
            print("\nExiting.... Thankyou!")
            break
        else:
            print("InvLid Option. Try Again!\n")  

if __name__ == "__main__":
    main()                          

File: calculator.c
#include <stdio.h>
#include <math.h>

// Function declarations
double calculateAcceleration(double v, double u, double t);
double calculateVelocity(double u, double a, double t);
double calculateDisplacement(double u, double t, double a);
double calculateForce(double m, double a);
double calculateKineticEnergy(double m, double v);
double calculatePotentialEnergy(double m, double g, double h);
double calculateWork(double f, double d);
double calculatePower(double w, double t);
double calculateMomentum(double m, double v);
double calculateFrequency(double T);
double evaluateExpression(double a, double b, double c); // (a + b) * c
double calculateSinTheta(double theta);
double calculateCosTheta(double theta);

int main() {
    int choice;
    double u, v, a, t, m, f, d, g = 9.8, h, w, T, theta, b, c;

    do {
        printf("\n--- Formula Solver Menu ---\n");
        printf("1. Acceleration (a = (v - u) / t)\n");
        printf("2. Final Velocity (v = u + a * t)\n");
        printf("3. Displacement (s = ut + 0.5at^2)\n");
        printf("4. Force (F = m * a)\n");
        printf("5. Kinetic Energy (KE = 0.5 * m * v^2)\n");
        printf("6. Potential Energy (PE = m * g * h)\n");
        printf("7. Work (W = F * d)\n");
        printf("8. Power (P = W / t)\n");
        printf("9. Momentum (p = m * v)\n");
        printf("10. Frequency (f = 1 / T)\n");
        printf("11. Expression: (a + b) * c\n");
        printf("12. Sin(theta)\n");
        printf("13. Cos(theta)\n");
        printf("0. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter initial velocity (u): "); scanf("%lf", &u);
                printf("Enter final velocity (v): "); scanf("%lf", &v);
                printf("Enter time (t): "); scanf("%lf", &t);
                printf("Acceleration = %.2lf\n", calculateAcceleration(v, u, t));
                break;
            case 2:
                printf("Enter initial velocity (u): "); scanf("%lf", &u);
                printf("Enter acceleration (a): "); scanf("%lf", &a);
                printf("Enter time (t): "); scanf("%lf", &t);
                printf("Final Velocity = %.2lf\n", calculateVelocity(u, a, t));
                break;
            case 3:
                printf("Enter initial velocity (u): "); scanf("%lf", &u);
                printf("Enter time (t): "); scanf("%lf", &t);
                printf("Enter acceleration (a): "); scanf("%lf", &a);
                printf("Displacement = %.2lf\n", calculateDisplacement(u, t, a));
                break;
            case 4:
                printf("Enter mass (m): "); scanf("%lf", &m);
                printf("Enter acceleration (a): "); scanf("%lf", &a);
                printf("Force = %.2lf\n", calculateForce(m, a));
                break;
            case 5:
                printf("Enter mass (m): "); scanf("%lf", &m);
                printf("Enter velocity (v): "); scanf("%lf", &v);
                printf("Kinetic Energy = %.2lf\n", calculateKineticEnergy(m, v));
                break;
            case 6:
                printf("Enter mass (m): "); scanf("%lf", &m);
                printf("Enter height (h): "); scanf("%lf", &h);
                printf("Potential Energy = %.2lf\n", calculatePotentialEnergy(m, g, h));
                break;
            case 7:
                printf("Enter force (F): "); scanf("%lf", &f);
                printf("Enter distance (d): "); scanf("%lf", &d);
                printf("Work = %.2lf\n", calculateWork(f, d));
                break;
            case 8:
                printf("Enter work (W): "); scanf("%lf", &w);
                printf("Enter time (t): "); scanf("%lf", &t);
                printf("Power = %.2lf\n", calculatePower(w, t));
                break;
            case 9:
                printf("Enter mass (m): "); scanf("%lf", &m);
                printf("Enter velocity (v): "); scanf("%lf", &v);
                printf("Momentum = %.2lf\n", calculateMomentum(m, v));
                break;
            case 10:
                printf("Enter time period (T): "); scanf("%lf", &T);
                printf("Frequency = %.2lf\n", calculateFrequency(T));
                break;
            case 11:
                printf("Enter value for a: "); scanf("%lf", &a);
                printf("Enter value for b: "); scanf("%lf", &b);
                printf("Enter value for c: "); scanf("%lf", &c);
                printf("Result = %.2lf\n", evaluateExpression(a, b, c));
                break;
            case 12:
                printf("Enter angle in degrees: "); scanf("%lf", &theta);
                printf("Sin(%.2lf°) = %.4lf\n", theta, calculateSinTheta(theta));
                break;
            case 13:
                printf("Enter angle in degrees: "); scanf("%lf", &theta);
                printf("Cos(%.2lf°) = %.4lf\n", theta, calculateCosTheta(theta));
                break;
            case 0:
                printf("Exiting program.\n");
                break;
            default:
                printf("Invalid choice! Please try again.\n");
        }
    } while (choice != 0);

    return 0;
}

// Function definitions
double calculateAcceleration(double v, double u, double t) { return (v - u) / t; }
double calculateVelocity(double u, double a, double t) { return u + a * t; }
double calculateDisplacement(double u, double t, double a) { return u * t + 0.5 * a * t * t; }
double calculateForce(double m, double a) { return m * a; }
double calculateKineticEnergy(double m, double v) { return 0.5 * m * v * v; }
double calculatePotentialEnergy(double m, double g, double h) { return m * g * h; }
double calculateWork(double f, double d) { return f * d; }
double calculatePower(double w, double t) { return w / t; }
double calculateMomentum(double m, double v) { return m * v; }
double calculateFrequency(double T) { return 1.0 / T; }
double evaluateExpression(double a, double b, double c) { return (a + b) * c; }
double calculateSinTheta(double theta) { return sin(theta * m_pi / 180.0); }
double calculateCosTheta(double theta) { return cos(theta * M_PI / 180.0); }
